platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -13,14 +13,25 @@ @interface MGLBackgroundStyleLayer ()
 @property (nonatomic) mbgl::style::BackgroundLayer *layer;
 @property (nonatomic, readwrite) NSString *layerIdentifier;
 @property (nonatomic, readwrite) NSString *sourceIdentifier;
+@property (nonatomic, readwrite) NSString *sourceLayerIdentifier;
 
 @end
 
 @implementation MGLBackgroundStyleLayer
 
 @synthesize mapView;
 
-- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier sourceIdentifier:(NSString *)sourceIdentifier {
+- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier
+{
+    if (self = [super init]) {
+        _layerIdentifier = layerIdentifier;
+        _layer = new mbgl::style::BackgroundLayer(layerIdentifier.UTF8String);
+    }
+    return self;
+}
+
+- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier sourceIdentifier:(NSString *)sourceIdentifier
```

Comment:
This is done in https://github.com/mapbox/mapbox-gl-native/pull/6269/commits/939cca1a3099b0598e2cdd9276a46790bbb9ed11